spec/models/search_impression_spec.rb

```diff
@@ -1,30 +1,42 @@
     end
-  end
 
-  context 'headers contains X-Original-Request header' do
-    let(:params) { { "foo" => "yep", "bar.blat" => "nope" } }
-    let(:search) { double(Search, modules: %w(BWEB), diagnostics: { AWEB: { snap: 'judgement' } }) }
-    let(:request) { double("request", remote_ip: '1.2.3.4', url: 'http://www.gov.gov/', referer: 'http://www.gov.gov/ref', user_agent: 'whatevs', headers: { 'X-Original-Request' => 'http://test.gov' }) }
-
-    it 'should log a non-null value for original_request' do
-      time = Time.now
-      allow(Time).to receive(:now).and_return time
-      expect(Rails.logger).to receive(:info).with("[X-Original-Request] (\"http://test.gov\")")
-      expect(Rails.logger).to receive(:info).with("[Search Impression] {\"clientip\":\"1.2.3.4\",\"request\":\"http://test.gov\",\"referrer\":\"http://www.gov.gov/ref\",\"user_agent\":\"whatevs\",\"diagnostics\":[{\"snap\":\"judgement\",\"module\":\"AWEB\"}],\"time\":\"#{time.to_formatted_s(:db)}\",\"vertical\":\"web\",\"modules\":\"BWEB\",\"params\":{\"foo\":\"yep\"}}")
-      SearchImpression.log(search, "web", params, request)
+    context 'with regular params' do
+      it 'has the expected log line' do
+        expect(Rails.logger).to have_received(:info).with("[Search Impression] {\"clientip\":\"1.2.3.4\",\"request\":\"http://www.gov.gov/\",\"referrer\":\"http://www.gov.gov/ref\",\"user_agent\":\"whatevs\",\"diagnostics\":[{\"snap\":\"judgement\",\"module\":\"AWEB\"}],\"time\":\"#{time.to_formatted_s(:db)}\",\"vertical\":\"web\",\"modules\":\"BWEB\",\"params\":{\"foo\":\"yep\"}}")
+      end
+    end
+
+    context 'params contains key with period' do
+      let(:params) { { 'foo' => 'yep', 'bar.blat' => 'nope' } }
+
+      it 'omits that parameter' do
+        expect(Rails.logger).to have_received(:info).with("[Search Impression] {\"clientip\":\"1.2.3.4\",\"request\":\"http://www.gov.gov/\",\"referrer\":\"http://www.gov.gov/ref\",\"user_agent\":\"whatevs\",\"diagnostics\":[{\"snap\":\"judgement\",\"module\":\"AWEB\"}],\"time\":\"#{time.to_formatted_s(:db)}\",\"vertical\":\"web\",\"modules\":\"BWEB\",\"params\":{\"foo\":\"yep\"}}")
+      end
+    end
+
+    context 'headers contains X-Original-Request header' do
+      let!(:request) { double('request', remote_ip: '1.2.3.4', url: 'http://www.gov.gov/', referer: 'http://www.gov.gov/ref', user_agent: 'whatevs', headers: { 'X-Original-Request' => 'http://test.gov' }) }
+
+      it 'should log a non-null value for original_request' do
+        expect(Rails.logger).to have_received(:info).with('[X-Original-Request] ("http://test.gov")')
+        expect(Rails.logger).to have_received(:info).with("[Search Impression] {\"clientip\":\"1.2.3.4\",\"request\":\"http://test.gov\",\"referrer\":\"http://www.gov.gov/ref\",\"user_agent\":\"whatevs\",\"diagnostics\":[{\"snap\":\"judgement\",\"module\":\"AWEB\"}],\"time\":\"#{time.to_formatted_s(:db)}\",\"vertical\":\"web\",\"modules\":\"BWEB\",\"params\":{\"foo\":\"yep\"}}")
+      end
```

Comment:
I realize these lines were in this spec already, so I'll leave this cleanup to your discretion. But if you *are* in a cleanup mood, I would wrap these and the other long lines in this spec:
```ruby
      let(:request) do
        double('request',
               remote_ip: '1.2.3.4',
               url: 'http://www.gov.gov/',
               referer: 'http://www.gov.gov/ref',
               user_agent: 'whatevs',
               headers: { 'X-Original-Request' => 'http://test.gov' })
      end

      it 'should log a non-null value for original_request' do
        expect(Rails.logger).to have_received(:info).
          with('[X-Original-Request] ("http://test.gov")')
        expect(Rails.logger).to have_received(:info).
          with('[X-Original-Request] ("http://test.gov")')
        expect(Rails.logger).to have_received(:info).with(
          "[Search Impression] {\"clientip\":\"1.2.3.4\",\"request\":\"http://test.gov\",\"referrer\":\"http://www.gov.gov/ref\",\"user_agent\":\"whatevs\",\"diagnostics\":[{\"snap\":\"judgement\",\"module\":\"AWEB\"}],\"time\":\"#{time.to_formatted_s(:db)}\",\"vertical\":\"web\",\"modules\":\"BWEB\",\"params\":{\"foo\":\"yep\"}}"
        )
      end
```